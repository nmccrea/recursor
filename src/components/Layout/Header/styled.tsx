import styled from "styled-components"
import { color, typography } from "styles/utilities"
import { Panel } from "components/parts/Panel"

export const H1 = styled.h1`
  ${typography.fontFamilyHeading}
  color: ${color.primary};
  text-transform: uppercase;
  line-height: 0;
  padding: 0.5rem 1rem;
`

export const HeaderPanel = styled(Panel)`
  position: fixed;
  top: 1rem;
  left: 1rem;
  padding: 0;
`
