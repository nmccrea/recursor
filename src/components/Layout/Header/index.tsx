import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import { Panel } from "components/parts/Panel"
import { HiddenContainer, H1, GithubImage } from "./styled"

const Header = () => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            githubUrl
          }
        }
      }
    `
  )

  return (
    <HiddenContainer>
      <Panel>
        <H1>{site.siteMetadata.title}</H1>
      </Panel>

      <a href={site.siteMetadata.githubUrl} target="blank">
        <GithubImage />
      </a>
    </HiddenContainer>
  )
}

export default Header
