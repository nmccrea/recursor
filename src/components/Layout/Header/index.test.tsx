import { render } from "@testing-library/react"
import "jest-styled-components"
import React from "react"
import Header from "."

describe("`<Header />`", () => {
  it("renders correctly", () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
