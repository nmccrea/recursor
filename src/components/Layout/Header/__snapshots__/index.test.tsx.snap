// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<Header />\` renders correctly 1`] = `
.c1 {
  font-family: "Audiowide",system-ui,-apple-system,BlinkMacSystemFont, "Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif, "Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  color: cornflowerblue;
  text-transform: uppercase;
  line-height: 1;
  margin: 0;
}

.c0 {
  padding: 1rem;
  background-color: rgba(255,255,255,0.7);
  border-radius: 0.5rem;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  position: fixed;
  top: 1rem;
  left: 1rem;
}

<div>
  <div
    class="sc-AxjAm c0"
  >
    <h1
      class="c1"
    >
      Mock Site Name
    </h1>
  </div>
</div>
`;
